src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -449,11 +523,13 @@ citus_shard_cost_by_disk_size(PG_FUNCTION_ARGS)
 	uint32 connectionFlag = 0;
 	PGresult *result = NULL;
 	bool raiseErrors = true;
-	char *sizeQuery = PG_TOTAL_RELATION_SIZE_FUNCTION;
+
 	ShardInterval *shardInterval = LoadShardInterval(shardId);
-	List *colocatedShardList = ColocatedShardIntervalList(shardInterval);
-	StringInfo tableSizeQuery = GenerateSizeQueryOnMultiplePlacements(colocatedShardList,
-																	  sizeQuery);
+	char *shardName = ConstructQualifiedShardName(shardInterval);
```

Comment:
Shouldn't we need to call `citus_partitioned_shard_total_size()` only for partitioned tables? And, also, shouldn't we need to skip partitions on the `ColocatedShardIntervalList` ?

Now, we assume all the `ColocatedShardIntervalList` contains only the partitions of `shardInterval`?